src/Core/File/Exception/FileNotFoundException.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\File\Exception;
+
+/**
+ * Thrown when file is not found
+ *
+ * @todo: Core/Exception/FileNotFoundExists. Is it worth introducing new one?
```

Comment:
`FileNotFoundException` was introduced early and is quite generic, but we couldn't know exactly where we were going at the time
So maybe this one should be the new exception to use, then the existing `FileNotFoundException` can extend this one and be declared as deprecated And we replace all its usages by this exception

Reply:
this deserves separate PR as there are lots of usage of that exception. I've added todo.
Also I think we need @matks approval for that

Reply:
It's a bit unrelated, but I would like to highlight a possible domain naming issue:

`src/Core/Exception/FileNotFound` is actually a "File On Disk Cannot be found". It's a very generic exception used for example when Translation Catalog file cannot be found or Theme file cannot be found. It refers to file as "computer resource for recording data".

We might have other File Exceptions where it would refer to file as "An attachment that can be assigned to a Virtual Product, and can be downloaded when the product has been purchased".

So let's be careful when using File 😅  (naming is hard !)

Apart from that 😄  I agree with @jolelievre let's move this exception by into the right folder

Reply:
I get it and I think we are good here.
Core/File is indeed related to filesystem, whereas for example Core/Domain/*/VirtualProductNotFoundException is all about domain, meaning that the `ProductDownload`(now named as `ProductFile`) is missing (e.g. not found in database).